index.js

```diff
@@ -65,8 +64,43 @@ function MLLPServer(host, port) {
 
     });
 
-    Server.listen(PORT, HOST);
+    self.send = function (receivingHost, receivingPort, hl7Data, callback) {
+        var sendingClient = new net.connect({
+            host: receivingHost,
+            port: receivingPort
+        }, function () {
+            console.log('Sending data to ' + receivingHost + ':' + receivingPort);
+            sendingClient.write(VT + hl7Data + FS + CR);
+        });
+
+        sendingClient.on('data', function (rawAckData) {
+            console.log(receivingHost + ':' + receivingPort + ' ACKED data');
+
+            var ackData = rawAckData
+                .toString() // Buffer -> String
+                .replace(VT, '')
+                .split('\r')[1] // Ack data
+                .replace(FS, '')
+                .replace(CR, '');
+
+            callback(null, ackData);
```

Comment:
@Eriatolc,

I'm not as bothered by having multiple patterns because they can be justified (which is my main rule in general with design patterns). I think we are okay moving forward with having these two patterns in the codebase.

> (Promises > callback > *).

I used a "nodeback" style callback so at least it's relatively familiar to most users. However, I would be more than happy to bring in the Q promise library so that we can take advantage of `.then` and `.catch`. Before I do this, are you okay with me pulling in another dependency (I mentioned Q, but Bluebird is just as good)?

> You must let the client app get the information of the ACK message

This is not an issue because the client will just use the callback like so:

```javascript
server.send('127.0.0.1', 1234, hl7, function (err, ackData) {
...
```

The issue is whether I should inspect the improve the ACK generation code and inspect the ACK message to determine if an error should be sent. Here's what the spec says:

> if the application processes the message successfully, it will generate the functional response message with a value of AA in acknowledgment code (MSA-1);

> or the application will send an error response, with a value of AE in acknowledgment code (MSA-1);

> if the application fails to process (reject) the message because of system uptime or other reasons and not for format or validation errors. The response message contains a value of AR in acknowledgment code (MSA-1). The message can sometimes be re-sent later but that is up to the sending system and the implementation to figure out.

The codebase currently calls `ackn(data2, "AA");` so that will need to be improved (this is the hard part and is probably outside of this PR but I will volunteer to fix it to conform to the spec). Adding a check for `AE` and `AR` and returning a failure will be easy.

Do you agree?

- Matthew